src/snowflake/connector/compute_chunk_size.py

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+constants = {
+    "CURRENT_CHUNK_SIZE": 8388608,
+    "MAX_OBJECT_SIZE": 5497558138880,
+    "MAX_PART_SIZE": 5368709120,
+    "MIN_PART_SIZE": 5242880,
+    "MAX_PARTS": 10000,
+}
+
+
+class ChunkSizeCalculator:
+    def __init__(
+        self,
+        current_chunk_size: int = constants["CURRENT_CHUNK_SIZE"],
+        max_object_size: int = constants["MAX_OBJECT_SIZE"],
+        max_part_size: int = constants["MAX_PART_SIZE"],
+        min_part_size: int = constants["MIN_PART_SIZE"],
+        max_parts: int = constants["MAX_PARTS"],
+    ) -> None:
+        self.current_chunk_size = current_chunk_size
+        self.max_object_size = max_object_size
+        self.max_part_size = max_part_size
+        self.min_part_size = min_part_size
+        self.max_parts = max_parts
+
+    def compute_chunk_size(self, file_size: int = None) -> int:
+        chunk_size = self.current_chunk_size
+        # check if we don"t exceed the allowed max file size 5 TiB
+        if file_size is not None and file_size < self.max_object_size:
+            chunk_size = self._check_max_parts(chunk_size, file_size)
+        else:
+            error_message = f"File size {file_size} exceeds the maximum file size {self.max_object_size}."
+            logger.error(error_message)
+            raise Exception(error_message)
```

Comment:
when users hit the error, what are they supposed to do with the error?

Reply:
The maximum file size we can upload via S3 per request is 5 TiB, customers would have to split their files over this limit to upload them in multiple requests.

Reply:
We should not raise the base class `Exception` and use something like `RuntimeError` at least.

Reply:
@sfc-gh-abenzaoui you can raise the basic connector error: https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/errors.py#L32